File: Shapes/shponimg.cpp
#include<cv.h>
#include<highgui.h>
#define w 250
using namespace cv;
void drawEllipse(Mat, double);
void drawCircle(Mat);
int main() {

	Mat image = imread("/home/darshil/images/smaller.jpg");

	Scalar color1;
	//four ellipses with angles 180deg, 135deg, 90deg, 45deg
	drawEllipse(image, 45);
	drawEllipse(image, 90);
	drawEllipse(image, 180);
	drawEllipse(image, 135);
	drawCircle(image);
	namedWindow("Display Image", CV_WINDOW_AUTOSIZE);
	imshow("Display Image", image);
	imwrite("/home/darshil/workspace/opennew1/src/images/Ellipses_Image.jpg",
			image);
	waitKey(0);

	return 0;
}
void drawEllipse(Mat img, double angle) {
	int thickness = 2;
	int lineType = 8;

	ellipse(img, Point(w / 2, w / 2), Size(w / 8, w / 32), angle, 0, 360,
			Scalar(255, 0, 0), thickness, lineType);
}

void drawCircle(Mat img) {
	int thickness = -1;
	int lineType = 8;
	circle(img, Point(w / 2, w / 2), w / 4, Scalar(255, 255, 200), thickness,
			lineType);

}

File: Shapes/shponimg.py
from cv2.cv import * # importing all the opencv libraries
img=LoadImageM("/home/darshil/images/qrcode.png") # reading image
Ellipse(img,(img.rows/2,img.cols/2),(100,25),0,0,360,(50,50,0),5) # Ellipse(img, center,axes,angle,start_angle,end_angle, color [, thickness [, lineType [, shift]]])
Ellipse(img,(img.rows/2,img.cols/2),(100,25),45,0,360,(255,255,255),-1) 
Ellipse(img,(img.rows/2,img.cols/2),(100,25),90,0,360,(50,0,50),5) 
Ellipse(img,(img.rows/2,img.cols/2),(100,25),135,0,360,(255,255,255),-1)
Circle(img,(img.rows/2,img.cols/2),50,(0,255,0),-1)
ShowImage("Window",img) # Displaying the image ... 
WaitKey(0)

